### PR TITLE
sort: Add a GNU-style Random Sorter

### PR DESCRIPTION
**Pull request by electricboogie**
The previous version was missing a GNU-style random sorter/shuffle ("-R").  I've implemented such a sorter and welcome your review.

### CODE REVIEW

File: src/uu/sort/src/sort.rs [hunk 1]
```diff
@@ -501,6 +534,31 @@ fn human_numeric_size_compare(a: &str, b: &str) -> Ordering {
...
+    da.cmp(&db)
+}
+
+fn get_rand_string() -> String {
+    let rand_string = thread_rng()
+        .sample_iter(&Alphanumeric)
+        .take(16)
+        .map(char::from)
+        .collect::<String>();
+    rand_string
```
Comment (sylvestre) [inline src/uu/sort/src/sort.rs:553]: probably enough ?

```suggestion
    thread_rng()
        .sample_iter(&Alphanumeric)
        .take(16)
        .map(char::from)
        .collect::<String>()
```

File: src/uu/sort/src/sort.rs [hunk 2]
```diff
@@ -12,6 +12,10 @@ extern crate uucore;
 
 use clap::{App, Arg};
 use itertools::Itertools;
+use twox_hash::XxHash64;
+use std::hash::{Hash, Hasher};
+use rand::distributions::Alphanumeric;
+use rand::{thread_rng, Rng};
```
Comment (sylvestre) [inline src/uu/sort/src/sort.rs:18]: please run rustfmt

File: src/uu/sort/src/sort.rs [hunk 3]
```diff
@@ -419,7 +448,11 @@ fn compare_by(a: &str, b: &str, settings: &Settings) -> Ordering {
     };
 
     for compare_fn in &settings.compare_fns {
-        let cmp = compare_fn(a, b);
+        let cmp: Ordering = if settings.random {
+            random_shuffle(a, b, settings.salt.clone())
```
Comment (sylvestre) [inline src/uu/sort/src/sort.rs:452]: please add tests to have a proper code coverage

**Review by sylvestre — CHANGES_REQUESTED** [review]

**Review by sylvestre — CHANGES_REQUESTED** [review]
Almost done, well done.
* it is great to have unit test but could you please add integration tests in:
https://github.com/uutils/coreutils/blob/master/tests/by-util/test_sort.rs
* please run "cargo +1.40.0 update" and amend Cargo.lock?

thanks

**Comment by electricboogie** [comment]
Pleased to add integration tests.  Thanks.

File: src/uu/sort/src/sort.rs [hunk 4]
```diff
@@ -29,15 +34,18 @@ static OPT_HUMAN_NUMERIC_SORT: &str = "human-numeric-sort";
 static OPT_MONTH_SORT: &str = "month-sort";
 static OPT_NUMERIC_SORT: &str = "numeric-sort";
 static OPT_VERSION_SORT: &str = "version-sort";
+static OPT_RANDOM_SORT: &str = "random-sort";
```
Comment (sylvestre) [inline src/uu/sort/src/sort.rs:37]:
```suggestion
```
Seems unused?

File: tests/by-util/test_sort.rs [hunk 5]
```diff
@@ -2,22 +2,43 @@ use crate::common::util::*;
 
 #[test]
 fn test_numeric_floats_and_ints() {
-    test_helper("numeric_floats_and_ints", "-n");
+    for numeric_sort_param in vec!["-n", "--numeric-sort"] {
```
Comment (sylvestre) [inline tests/by-util/test_sort.rs:5]: please run rustfmt

**Comment by sylvestre** [comment]
```
 codecov/project Successful in 1s — 63.49% (+0.17%) compared to 0bdd61a 
```
Bravo :)

**Comment by sylvestre** [comment]
Please rebase Cargo.lock  and it should be good :)

**Pull request merged**